ModuleNotFoundError: No module named 'async_timeout'

Hi!

Thanks for all of your efforts maintaining this library.

`redis-py` no longer installs `async_timeout` on python 3.11 (https://github.com/redis/redis-py/pull/2602/files), which manifests in this error in our CI:

```
ImportError while loading conftest '/home/runner/work/starlite/starlite/tests/conftest.py'.
tests/conftest.py:54: in <module>
    from fakeredis.aioredis import FakeRedis
.venv/lib/python3.11/site-packages/fakeredis/aioredis.py:12: in <module>
    import async_timeout
E   ModuleNotFoundError: No module named 'async_timeout'
```

Reproduction:

```
peter@peter-Inspiron-15-5510:~/PycharmProjects/starlite$ pyenv shell 3.11.2
peter@peter-Inspiron-15-5510:~/PycharmProjects/starlite$ python -m venv .venv
peter@peter-Inspiron-15-5510:~/PycharmProjects/starlite$ . .venv/bin/activate
(.venv) peter@peter-Inspiron-15-5510:~/PycharmProjects/starlite$ python --version
Python 3.11.2
(.venv) peter@peter-Inspiron-15-5510:~/PycharmProjects/starlite$ python -m pip install fakeredis
Collecting fakeredis
  Using cached fakeredis-2.10.1-py3-none-any.whl (57 kB)
Collecting redis<5,>=4
  Downloading redis-4.5.2-py3-none-any.whl (238 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 238.5/238.5 kB 5.4 MB/s eta 0:00:00
Collecting sortedcontainers<3.0,>=2.4
  Using cached sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)
Installing collected packages: sortedcontainers, redis, fakeredis
Successfully installed fakeredis-2.10.1 redis-4.5.2 sortedcontainers-2.4.0

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: pip install --upgrade pip
(.venv) peter@peter-Inspiron-15-5510:~/PycharmProjects/starlite$ python -c "from fakeredis.aioredis import FakeRedis"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/peter/PycharmProjects/starlite/.venv/lib/python3.11/site-packages/fakeredis/aioredis.py", line 12, in <module>
    import async_timeout
ModuleNotFoundError: No module named 'async_timeout'
```

As `fakeredis-py` directly depends on `async_timeout` perhaps you can consider adding it to the lib's dependencies. Alternatively, you could go all-in on the transitive dependency and import it from `redis-py`, e.g., `from redis.asyncio.connection import async_timeout`, and leverage their version specific aliasing.

--- a/fakeredis/aioredis.py
+++ b/fakeredis/aioredis.py
@@ -9,7 +9,11 @@ if sys.version_info >= (3, 8):
 else:
     from typing_extensions import Type, TypedDict
 
-import async_timeout
+if sys.version_info >= (3, 11):
+    from asyncio import timeout as async_timeout
+else:
+    from async_timeout import timeout as async_timeout
+
 import redis.asyncio as redis_async  # aioredis was integrated into redis in version 4.2.0 as redis.asyncio
 
 from . import _fakesocket, FakeServer

--- a/fakeredis/aioredis.py
+++ b/fakeredis/aioredis.py
@@ -33,7 +33,7 @@ class AsyncFakeSocket(_fakesocket.FakeSocket):
     async def _async_blocking(self, timeout, func, event, callback):
         result = None
         try:
-            async with async_timeout.timeout(timeout if timeout else None):
+            async with async_timeout(timeout if timeout else None):
                 while True:
                     await event.wait()
                     event.clear()

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -63,7 +63,6 @@ types-redis = "^4"
 twine = "^4.0" # Upload to pypi
 tox = "^4.2.6"
 tox-docker = "^4"
-async_timeout = "^4"
 
 [tool.poetry.urls]
 "Bug Tracker" = "https://github.com/cunla/fakeredis-py/issues"

--- a/test/test_redis_asyncio.py
+++ b/test/test_redis_asyncio.py
@@ -125,7 +125,7 @@ async def test_pubsub(req_aioredis2, event_loop):
                     break
                 queue.put_nowait(message)
 
-    async with async_timeout.timeout(5), req_aioredis2.pubsub() as ps:
+    async with async_timeout(5), req_aioredis2.pubsub() as ps:
         await ps.subscribe('channel')
         task = event_loop.create_task(reader(ps))
         await req_aioredis2.publish('channel', 'message1')

--- a/test/test_redis_asyncio.py
+++ b/test/test_redis_asyncio.py
@@ -1,7 +1,11 @@
 import asyncio
 import re
+import sys
 
-import async_timeout
+if sys.version_info >= (3, 11):
+    from asyncio import timeout as async_timeout
+else:
+    from async_timeout import timeout as async_timeout
 import pytest
 import pytest_asyncio
 import redis

